packages/opentelemetry-plugin-document-load/src/documentLoad.ts

```diff
@@ -0,0 +1,229 @@
+import { AttributeNames } from './enums/AttributeNames';
+import { PerformanceTimingNames as PTN } from './enums/PerformanceTimingNames';
+import { PerformanceEntries, PerformanceLegacy } from './types';
+import { hasKey } from './utils';
+
+/**
+ * This class represents a document load plugin
+ */
+export class DocumentLoad extends BasePlugin<unknown> {
+  readonly component: string = 'document-load';
+  readonly version: string = '1';
+  moduleName = this.component;
+  protected _config!: PluginConfig;
+
+  /**
+   *
+   * @param config
+   */
+  constructor(config: PluginConfig = {}) {
+    super();
+    this._onDocumentLoaded = this._onDocumentLoaded.bind(this);
+    this._config = config;
+  }
+
+  /**
+   * callback to be executed when page is loaded
+   */
+  private _onDocumentLoaded() {
+    this._collectPerformance();
+  }
+
+  /**
+   * Helper function for starting an event
+   * @param span
+   * @param performanceName name of performance entry for time start
+   * @param entries
+   */
+  private _addSpanEvent(
+    span: Span,
+    performanceName: string,
+    entries: PerformanceEntries
+  ): Span | undefined {
+    if (
+      hasKey(entries, performanceName) &&
+      typeof entries[performanceName] === 'number'
+    ) {
+      span.addEvent(performanceName, undefined, entries[performanceName]);
+      return span;
+    }
+    return undefined;
+  }
+
+  /**
+   * Collects information about performance and creates appropriate spans
+   */
+  private _collectPerformance() {
+    const entries = this._getEntries();
+
+    const rootSpan = this._startSpan(
+      AttributeNames.DOCUMENT_LOAD,
+      PTN.FETCH_START,
+      entries
+    );
+    if (!rootSpan) {
+      return;
+    }
+    const fetchSpan = this._startSpan(
+      AttributeNames.DOCUMENT_FETCH,
+      PTN.FETCH_START,
+      entries,
+      {
+        parent: rootSpan,
+      }
+    );
+    if (fetchSpan) {
+      this._addSpanEvent(fetchSpan, PTN.DOMAIN_LOOKUP_START, entries);
+      this._addSpanEvent(fetchSpan, PTN.DOMAIN_LOOKUP_END, entries);
+      this._addSpanEvent(fetchSpan, PTN.CONNECT_START, entries);
+      this._addSpanEvent(fetchSpan, PTN.SECURE_CONNECTION_START, entries);
+      this._addSpanEvent(fetchSpan, PTN.CONNECT_END, entries);
+      this._addSpanEvent(fetchSpan, PTN.REQUEST_START, entries);
+      this._addSpanEvent(fetchSpan, PTN.RESPONSE_START, entries);
+
+      this._endSpan(fetchSpan, PTN.RESPONSE_END, entries);
+    }
+
+    this._addSpanEvent(rootSpan, PTN.UNLOAD_EVENT_START, entries);
+    this._addSpanEvent(rootSpan, PTN.UNLOAD_EVENT_END, entries);
+    this._addSpanEvent(rootSpan, PTN.DOM_INTERACTIVE, entries);
+    this._addSpanEvent(rootSpan, PTN.DOM_CONTENT_LOADED_EVENT_START, entries);
+    this._addSpanEvent(rootSpan, PTN.DOM_CONTENT_LOADED_EVENT_END, entries);
+    this._addSpanEvent(rootSpan, PTN.DOM_COMPLETE, entries);
+
+    this._endSpan(rootSpan, PTN.LOAD_EVENT_START, entries);
```

Comment:
Would it make sense to use load event end instead and have load event start be a timed event?

See https://developer.mozilla.org/en-US/docs/Web/API/PerformanceNavigationTiming/loadEventEnd